File: app/src/main/java/com/i14yokoro/mldpterminal/BleDevice.java
package com.i14yokoro.mldpterminal;

import android.support.annotation.NonNull;

/**
 * デバイスを表すクラス
 */
class BleDevice {
    private final String address;
    private final String name;

    BleDevice(String address, String name) {
        this.address = address;
        this.name = name;
    }

    public String getAddress() {
        return address;
    }

    public String getName() {
        return name;
    }

    @NonNull
    @Override
    public String toString(){
        String str;
        str = "Name: " + this.name  + "   --address: " + this.address;
        return str;
    }

    @Override
    public boolean equals(Object object) {
        if (object instanceof BleDevice) {
            return this.address.equals(((BleDevice) object).address);
        }
        return false;
    }

    @Override
    public int hashCode() {
        return this.address.hashCode();
    }

}

File: app/src/main/java/com/i14yokoro/mldpterminal/bluetooth/BleDevice.kt
package com.i14yokoro.mldpterminal.bluetooth

/**
 * デバイスを表すクラス
 */
internal class BleDevice(val address: String?, val name: String?) {

    override fun toString(): String {
        return "Name: $name   --address: $address"
    }

    override fun equals(other: Any?): Boolean {
        return if (other is BleDevice) {
            this.address == other.address
        } else false
    }

    override fun hashCode(): Int {
        return this.address.hashCode()
    }

}

